### PR TITLE
Group 2

### PR DESCRIPTION
**Pull request by TehMillhouse**
Big oof.

### CODE REVIEW

File: src/main/java/de/cas/mse/exercise/csvdb/impl/AddressDb.java [hunk 1]
```diff
@@ -9,80 +9,68 @@
...
 import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 import de.cas.mse.exercise.csvdb.CsvDB;
+import de.cas.mse.exercise.csvdb.PersistenceLayer;
+import de.cas.mse.exercise.csvdb.Record;
 import de.cas.mse.exercise.csvdb.data.Address;
 
-public class AddressDb implements CsvDB<Address> {
+public class AddressDb<T extends PersistenceLayer> implements CsvDB<Address> {
```
Comment (corebonts) [inline src/main/java/de/cas/mse/exercise/csvdb/impl/AddressDb.java:19]: Do you really need generics? T could be PersistenceLayer on its own maybe.

File: src/main/java/de/cas/mse/exercise/csvdb/impl/AddressDb.java [hunk 2]
```diff
@@ -9,80 +9,68 @@
...
-		try {
-			final List<String> lines = Files.readAllLines(tableFile);
-			return lines.stream().map(e -> turnToAddress(e)).collect(Collectors.toList());
-		} catch (final IOException e) {
-			throw new RuntimeException(e);
-		}
-	}
+    private Address turnToAddress(final Record record) {
+        final Address addressObject = new Address();
+        addressObject.setGuid(record.fields[0]);
```
Comment (The-Drummer6) [inline src/main/java/de/cas/mse/exercise/csvdb/impl/AddressDb.java:39]: at this point you have to know which field has which index

File: src/main/java/de/cas/mse/exercise/csvdb/Record.java [hunk 3]
```diff
@@ -0,0 +1,5 @@
+package de.cas.mse.exercise.csvdb;
+
+public class Record {
+    public String[] fields;
```
Comment (zopad) [inline src/main/java/de/cas/mse/exercise/csvdb/Record.java:4]: What is the benefit of this abstraction?

Comment (TehMillhouse) [reply src/main/java/de/cas/mse/exercise/csvdb/Record.java:4]: I misunderstood the purpose of the exercise and thought we were supposed to make it work for representations other than CSV :|

File: src/main/java/de/cas/mse/exercise/csvdb/impl/AddressDb.java (hunk 1 above)
Comment (TehMillhouse) [reply src/main/java/de/cas/mse/exercise/csvdb/impl/AddressDb.java:19]: Right. Not on my A-game on this exercise.

**Pull request merged**